### PR TITLE
Improve error handling and migrate connector to RC1

### PR DESCRIPTION
**Pull request by dulanjalidilmi**
## Purpose
Fix https://github.com/wso2/ballerina-integrator/issues/49
Fix https://github.com/wso2/ballerina-integrator/issues/206

### CODE REVIEW

File: src/amazons3/Module.md [hunk 1]
```diff
@@ -2,12 +2,13 @@ Connects to Amazon S3 from Ballerina.
 
 # Module Overview
 
-The Amazon S3 client allows you to access the Amazon S3 REST API through ballerina. The following section provide you the details on connector operations.
-
+The Amazon S3 client allows you to access Amazon S3 REST API through ballerina. The following section provide you the
```
Comment (samgnaniah) [inline src/amazons3/Module.md]:
```suggestion
The Amazon S3 client allows you to access Amazon S3 REST API through Ballerina. The following section provides  
```

Better if we change "ballerina" to "Ballerina" in all instances whenever we see it.

File: src/amazons3/Module.md [hunk 2]
```diff
@@ -2,12 +2,13 @@ Connects to Amazon S3 from Ballerina.
...
-The Amazon S3 client allows you to access the Amazon S3 REST API through ballerina. The following section provide you the details on connector operations.
-
+The Amazon S3 client allows you to access Amazon S3 REST API through ballerina. The following section provide you the 
+details on connector operations.
 
 **Buckets Operations**
 
-The `wso2/amazons3` module contains operations that work with buckets. You can list the existing buckets, create a bucket, delete a bucket and list objects in a bucket.
+The `wso2/amazons3` module contains operations that work with buckets. You can list the existing buckets, create a bucket, 
+delete a bucket and list objects in a bucket.
```
Comment (samgnaniah) [inline src/amazons3/Module.md]:
```suggestion
delete a bucket, and list objects in a bucket.
```

Added serial comma. Normally we have a comma before "and" in a list.

File: src/amazons3/amazons3_errors.bal [hunk 3]
```diff
@@ -17,70 +17,149 @@
...
     string message;
-    string errorCode;
     error cause?;
 };
 
-// Ballerina AmazonS3 Client Error Types
-
-public const CLIENT_CONFIG_INITIALIZATION_FAILED = "{wso2/amazons3}ClientConfigInitializationFailed";
-public type ClientConfigInitializationFailed error<CLIENT_CONFIG_INITIALIZATION_FAILED, ErrorDetail>;
+// Client error reasons
```
Comment (samgnaniah) [inline src/amazons3/amazons3_errors.bal]: Not sure if reasons is the right word here. Is it types or reasons based on types?

Comment (dulanjalidilmi) [reply src/amazons3/amazons3_errors.bal]: These are reasons for respective client error types.

File: README.md [hunk 4]
```diff
@@ -6,13 +6,13 @@ The Amazon S3 client allows you to access the Amazon S3 REST API through balleri
 
 ## Compatibility
 | Ballerina Language Version | Amazon S3 API version  |
-| -------------------------- | -------------------- |
-| 0.992.0                    | 2006-03-01                  |
+| -------------------------- | ---------------------- |
+| 1.0.0-alpha3               | 2006-03-01             |
```
Comment (Maninda) [inline README.md]:
```suggestion
| 1.0.0               | 2006-03-01             |
```

File: README.md [hunk 5]
```diff
@@ -6,13 +6,13 @@ The Amazon S3 client allows you to access the Amazon S3 REST API through balleri
...
 
-|                    |    Version     |
-|:------------------:|:--------------:|
-| Ballerina Language |   0.992.0      |
-| Amazon S3 API      |   2006-03-01   |
+|                    |    Version           |
+|:------------------:|:--------------------:|
+| Ballerina Language |  1.0.0-rc1-SNAPSHOT  |
```
Comment (Maninda) [inline README.md]:
```suggestion
| Ballerina Language |  1.0.0               |
```

Comment (dulanjalidilmi) [reply README.md]: Fixed

File: src/amazons3/Module.md (hunk 1 above)
Comment (dulanjalidilmi) [reply src/amazons3/Module.md]: Fixed

File: src/amazons3/Module.md (hunk 2 above)
Comment (dulanjalidilmi) [reply src/amazons3/Module.md]: Fixed

File: README.md (hunk 4 above)
Comment (dulanjalidilmi) [reply README.md]: Fixed.

**Pull request merged**